river/cluster/textclust.py

```diff
@@ -0,0 +1,595 @@
+import math
+import typing
+
+import numpy as np
+import pandas as pd
+from sklearn.cluster import AgglomerativeClustering
+
+from river import base
+
+__all__ = ["textclust"]
+
+
+class textclust(base.Clusterer):
```

Comment:
It's more common in Python to use `PascalCase` for class names. So I would suggest calling this `TextClust`. Else some people are going to think this is a function.